lib/facter/self_service.rb

```diff
@@ -114,6 +114,81 @@
     { S0014: res.nil? }
   end
 
+  chunk(:S0016) do
+    # Puppetserver
+    next unless PuppetSelfService.primary? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler?
+    log_path = Puppet.settings['logdir'].to_s + '/../puppetserver/puppetserver.log'
+    line_count = `wc -l < #{log_path}`.to_i
+    lines_to_read = 250
```

Comment:
This code is repeated 3 times. We should move it to a method in the shared library. Maybe something that allows for passing in a `log`, `search_string`, and an optional `lines_to_read`.

Reply:
Will be working on this in later PR